### PR TITLE
Joda time config converters

### PR DESCRIPTION
**Pull request by queimadus**
Add `ConfigConvert`ers to `joda-time`'s `Instant`, `Period`, `Interval`, `Duration`, `DateTimeFormatter` and `DateTimeZone`.

Brought to you by [scala-swarm scala hackathon](https://www.scala-swarm.org/), courtesy of @queimadus and @andrebeat.

EDIT: Has the changes from the #268 suggestions.

### CODE REVIEW

**Review by derekmorr — APPROVED** [review]

File: modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.typesafe.config.{ ConfigFactory, ConfigValueFactory }
+import org.joda.time._
+import org.scalacheck.Arbitrary
+import JodaSpec._
+import org.joda.time.format.{ DateTimeFormat, DateTimeFormatter }
+import pureconfig.module.joda.configurable.ConfigurableSuite.{ dateTimeArbitrary, zoneIdArbitrary }
+import pureconfig.BaseSuite
+import pureconfig.syntax._
+
+class JodaSpec extends BaseSuite {
```
Comment (leifwickland) [inline modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala]: @queimadus Terribly sorry. I remembered the convention in PureConfig incorrectly. This should be renamed to `JodaSuite`. That's entirely my fault.

**Review by leifwickland — CHANGES_REQUESTED** [review]
@queimadus Would you be willing to fix my mistake? Sorry about that. 
You'll have my approval afterward.

File: modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  it should "parse Instant" in forAll {
```
Comment (ruippeixotog) [inline modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala]: This test and the ones below all test the reading part of the `ConfigConvert`, but not the writing part.

`BaseSuite` provides a convenient `checkArbitrary[A]` method that checks that writing an `A` to a config and reading it back should yield an `A` instance equal to the original one. Consider replacing this whole test by a single `checkArbitrary[Instant]` (same thing with the other types).

**Review by ruippeixotog — COMMENTED** [review]
Other than the minor comment below, LGTM! Thanks for this, @queimadus and @andrebeat.

File: modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala (hunk 1 above)
Comment (queimadus) [reply modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala]: Fixed in 92a6a96.

File: modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala (hunk 2 above)
Comment (queimadus) [reply modules/joda/src/test/scala/pureconfig/module/joda/JodaSpec.scala]: I've simplified all `joda` tests with `checkArbitrary` in 92a6a96.

**Review by ruippeixotog — APPROVED** [review]
LGTM!

**Review by leifwickland — APPROVED** [review]
Thanks, @queimadus and @andrebeat! I really appreciate your contribution and patience.

**Comment by jcazevedo** [comment]
Thanks for the PR @queimadus and @andrebeat! We would like to add you to the [CONTRIBUTORS](https://github.com/pureconfig/pureconfig/blob/master/CONTRIBUTORS) file. If you agree, please submit another PR in which you add a line with your names, your emails (optional) and your usernames, in line with PRs such as #118, #152 or #160.

**Pull request merged**